om4labs/diags/seaice/seaice.py

```diff
@@ -0,0 +1,449 @@
+        type=str,
+        default="March",
+        help="Month to analyze. Deafult is March",
+    )
+
+    parser.add_argument(
+        "-i",
+        "--interactive",
+        action="store_true",
+        help="Interactive mode displays plot to screen. Default is False",
+    )
+
+    parser.add_argument(
+        "-s", "--static", type=str, default=None, help="Path to static file"
+    )
+
+    parser.add_argument(
+        "-r",
+        "--region",
+        type=str,
+        default="nh",
+        help="Region for analysis. Default is nh. Options are nh and sh",
+    )
+
+    parser.add_argument(
+        "-O",
+        "--obsfile",
+        type=str,
+        default=None,
+        help="Path to file containing observations",
+    )
+
+    parser.add_argument(
+        "-l", "--label", type=str, default="", help="String label to annotate the plot"
+    )
+
+    return parser.parse_args(cliargs)
+
+
+def calc_seaice(infile, static, obsfile, label=None, region="nh"):
+    """Main computational script"""
+
+    # Open ice model output and the static file
+    ds = xr.open_mfdataset(infile, combine="by_coords")
+    if "siconc" in ds.variables:
+        ds["CN"] = ds["siconc"]
+        if ds["CN"].max() > 1.0:
+            ds["CN"] = ds["CN"] / 100.0
+    else:
+        ds["CN"] = ds["CN"].sum(dim="ct")
+    dstatic = xr.open_dataset(static)
+    ds["CELL_AREA"] = dstatic["CELL_AREA"]
+    ds["GEOLON"] = dstatic["GEOLON"]
+    ds["GEOLAT"] = dstatic["GEOLAT"]
+    ds["AREA"] = dstatic["CELL_AREA"] * 4.0 * np.pi * (6.378e6 ** 2)
+
+    # Get model start year and end year
+    modtime = (
+        int(ds["time"].isel({"time": 0}).dt.strftime("%Y")),
+        int(ds["time"].isel({"time": -1}).dt.strftime("%Y")),
+    )
+
+    # Get Valid Mask
+    valid_mask = np.where(ds["CELL_AREA"] == 0, True, False)
+
+    # Open observed SIC on 25-km EASE grid (coords already named lat and lon)
+    dobs = xr.open_dataset(obsfile)
+
+    # Get observation start year and end year
+    obstime = (
+        int(dobs["time"].isel({"time": 0}).dt.strftime("%Y")),
+        int(dobs["time"].isel({"time": -1}).dt.strftime("%Y")),
+    )
+
+    # Create annual cycle climatology
+    model = averagers.annual_cycle(ds, "CN")
+    obs = averagers.annual_cycle(dobs, "sic")
+
+    # Calculate area and extent
+    if region == "nh":
+        area_model = np.where(ds["GEOLAT"] > 0.0, model * ds.AREA, 0.0)
+        ext_model = np.where((model > 0.15) & (ds["GEOLAT"] > 0.0), ds.AREA, 0.0)
+    elif region == "sh":
+        area_model = np.where(ds["GEOLAT"] < 0.0, model * ds.AREA, 0.0)
+        ext_model = np.where((model > 0.15) & (ds["GEOLAT"] < 0.0), ds.AREA, 0.0)
+    else:
+        raise ValueError(f"Unknown region {region}. Option are nh or sh")
+    area_model = area_model.sum(axis=(-2, -1)) * 1.0e-12
+    ext_model = ext_model.sum(axis=(-2, -1)) * 1.0e-12
+
+    area_obs = obs * dobs.areacello
+    area_obs = area_obs.sum(axis=(-2, -1)) * 1.0e-12
```

Comment:
I'd use labels instead of axes index

Reply:
Dimension names can change across different versions of GFDL models.  If we want this to work on legacy models, e.g. ESM2*, CM2.1, etc. this might be better for now.  We'd have to write extra logic to detect the axis names.  I think this is safe for now.